blockchain/state_migration.go

```diff
@@ -491,31 +489,24 @@ func (bc *BlockChain) StartWarmUp() error {
 		return fmt.Errorf("target cache is nil")
 	}
 	db := state.NewDatabaseWithExistingCache(bc.db, cache)
+	children, err := db.TrieDB().NodeChildren(block.Root())
+	if err != nil {
+		return err
+	}
 
 	bc.quitWarmUp = make(chan struct{})
+	logger.Info("Warm up is started", "blockNum", block.NumberU64(), "root", block.Root().String(), "len(children)", len(children))
 
-	go func() {
-		defer func() {
-			bc.quitWarmUp = nil
-		}()
-
-		children, err := db.TrieDB().NodeChildren(block.Root())
-		if err != nil {
-			logger.Error("Cannot start warming up", "err", err)
-			return
-		}
-
-		logger.Info("Warm up is started", "blockNum", block.NumberU64(), "root", block.Root().String(), "len(children)", len(children))
-
-		resultHashCh := make(chan common.Hash, 10000)
-		resultErrCh := make(chan error)
-
-		for _, child := range children {
-			go bc.concurrentIterateTrie(child, db, resultHashCh, resultErrCh)
-		}
+	resultCh := make(chan struct{}, 10000)
+	errCh := make(chan error)
+	for _, child := range children {
+		go bc.concurrentIterateTrie(child, db, resultCh, errCh)
+	}
 
+	go func() {
```

Comment:
`for _, child := range` for loop is taken out from this go func. Is it ok?

Reply:
@yoomee1313 Yes, above code(=L#502~L#504) is to run child iterators with different go-routines and below is to receive result from those child go-routines and check if the task is done or not.

Reply:
@yoomee1313 As it is approved by 3 reviewers, let me merge this change for now. Please let me know if you have any issues 😄